### PR TITLE
Makefile and config cleanup and enhancements

### PR DESCRIPTION
**Pull request by Jafaral**
See individual commit msgs for details. Short summary:
* Drop legacy Makefile rules
* Fix git command call
* Give the user the option to require a feature when calling ./configure

### CODE REVIEW

File: Makefile
Comment (Don-Ward) [inline Makefile:9]: Typo here?

File: Makefile.in
Comment (Don-Ward) [inline Makefile.in:9]: Typo?

**Review by Don-Ward — APPROVED** [review]
The duplication of REPO_REV_... has gone (which is a good thing) but we still have the duplication of the update_rev target. Is it possible to refactor that also? No worries if "no" or "too hard". There is a small typo in Makefile.in (and Makefile) but it's in a comment and not a show stopper.

**Comment by Jafaral** [comment]
Will do update_rev as as well and push it to the same PR.

**Comment by Don-Ward** [comment]
It looks fine. I have one question, which is can we find a more neutral word than "dirty"? I considered the following alternatives (in order of preference, best first)
-modified
-local-edits
-edits
-altered
-editted

**Comment by brucerennie** [comment]
Is there any particular reason for wanting a "more neutral" word than "dirty"? Within the context of what is being done, it gives a very clear indication that something needs to be done as a priority. Whereas, the words suggested are mild in comparison, almost of no significance to the process.

**Comment by Don-Ward** [comment]
> Is there any particular reason for wanting a "more neutral" word than "dirty"? Within the context of what is being done, it gives a very clear indication that something needs to be done as a priority. Whereas, the words suggested are mild in comparison, almost of no significance to the process.

Well, to me (at least) the word "dirty" implies there is something wrong.  There's nothing wrong with building a system that has local modifications (which is all the --dirty flag is indicating) so I think that a more accurate description, which will be readily understood by people who aren't that familiar with git's terminology, is preferable.

I also disagree that "something needs to be done".  The real purpose is to alert the maintainers (if the user has a problem) that they will not be able to reproduce the the system that the user is running.

**Comment by brucerennie** [comment]
Thanks for the background to your thinking here. I don't know if it would change my thinking here, but I can understand where you are coming from. I still think that your selection of alternatives do not fulfil your purpose though. If you purpose is to alert the maintainers in this fashion, a more descriptive term would be better. Not that I can think of any at this point so I'll leave it to you and Jafar to make your decision. It is not affecting me personally at this time.

**Pull request merged**
